net/instaweb/rewriter/strip_subresource_hints_filter_test.cc

```diff
@@ -0,0 +1,234 @@
+                                    &message_handler_);
+    lawyer->AddRewriteDomainMapping(kTo2Domain, kFrom2Domain,
+                                    &message_handler_);
+    lawyer->AddShard(kTo2Domain, StrCat(kTo2ADomain, ",", kTo2BDomain),
+                     &message_handler_);
+    rewrite_driver()->AddFilters();
+  }
+};
+
+class StripSubresourceHintsFilterTestPartialPreserve : public RewriteTestBase {
+ protected:
+  virtual void SetUp() {
+    RewriteTestBase::SetUp();
+    options()->Disallow("*dontdropme*");
+    options()->set_css_preserve_urls(false);
+    options()->set_js_preserve_urls(false);
+    options()->set_image_preserve_urls(true);
+    DomainLawyer* lawyer = options()->WriteableDomainLawyer();
+    lawyer->AddRewriteDomainMapping(kTo1Domain, kFrom1Domain,
+                                    &message_handler_);
+    lawyer->AddRewriteDomainMapping(kTo2Domain, kFrom2Domain,
+                                    &message_handler_);
+    lawyer->AddShard(kTo2Domain, StrCat(kTo2ADomain, ",", kTo2BDomain),
+                     &message_handler_);
+    rewrite_driver()->AddFilters();
+  }
+};
+
+class StripSubresourceHintsFilterTestFullPreserve : public RewriteTestBase {
+ protected:
+  virtual void SetUp() {
+    RewriteTestBase::SetUp();
+    options()->Disallow("*dontdropme*");
+    options()->set_css_preserve_urls(true);
+    options()->set_js_preserve_urls(true);
+    options()->set_image_preserve_urls(true);
+    DomainLawyer* lawyer = options()->WriteableDomainLawyer();
+    lawyer->AddRewriteDomainMapping(kTo1Domain, kFrom1Domain,
+                                    &message_handler_);
+    lawyer->AddRewriteDomainMapping(kTo2Domain, kFrom2Domain,
+                                    &message_handler_);
+    lawyer->AddShard(kTo2Domain, StrCat(kTo2ADomain, ",", kTo2BDomain),
+                     &message_handler_);
+    rewrite_driver()->AddFilters();
+  }
+};
+
+class StripSubresourceHintsFilterTestDisabled : public RewriteTestBase {
+ protected:
+  virtual void SetUp() {
+    RewriteTestBase::SetUp();
+    options()->Disallow("*dontdropme*");
+    options()->set_preserve_subresource_hints(true);
+    DomainLawyer* lawyer = options()->WriteableDomainLawyer();
+    lawyer->AddRewriteDomainMapping(kTo1Domain, kFrom1Domain,
+                                    &message_handler_);
+    lawyer->AddRewriteDomainMapping(kTo2Domain, kFrom2Domain,
+                                    &message_handler_);
+    lawyer->AddShard(kTo2Domain, StrCat(kTo2ADomain, ",", kTo2BDomain),
+                     &message_handler_);
+    rewrite_driver()->AddFilters();
+  }
+};
+
+TEST_F(StripSubresourceHintsFilterTest, PreserveSubResourceHintsIsFalse) {
+  EXPECT_FALSE(options()->preserve_subresource_hints());
+}
+
+TEST_F(StripSubresourceHintsFilterTest, SingleResourceNoLink) {
+  ValidateExpected("single_resource_no_link",
+      "<head><link rel=\"subresource\"/></head>"
```

Comment:
no need to change your tests, but in the future you'd need to do less escaping if you used single quotes for the attributes